server/src/test/java/org/elasticsearch/indices/recovery/DanglingIndicesIT.java

```diff
@@ -0,0 +1,148 @@
+        logger.info("--> starting cluster");
+        final Settings settings = buildSettings(true);
+        internalCluster().startNodes(3, settings);
+
+        // Create an index and distribute it across the 3 nodes
+        createAndPopulateIndex(INDEX_NAME, SHARD_COUNT, REPLICA_COUNT);
+        ensureGreen();
+
+        // This is so that when then node comes back up, we have a dangling index that can be recovered.
+        logger.info("--> restarted a random node and deleting the index while it's down");
+        internalCluster().restartRandomDataNode(new InternalTestCluster.RestartCallback() {
+
+            @Override
+            public Settings onNodeStopped(String nodeName) throws Exception {
+                deleteIndex(INDEX_NAME);
+                return super.onNodeStopped(nodeName);
+            }
+        });
+
+        ensureGreen();
+
+        assertTrue("Expected dangling index " + INDEX_NAME + " to be recovered", indexExists(INDEX_NAME));
+    }
+
+    /**
+     * Check that when dangling indices are discovered, then they are not recovered into
+     * the cluster when the recovery setting is disabled.
+     */
+    public void testDanglingIndicesAreNotRecoveredWhenSettingIsDisabled() throws Exception {
+        logger.info("--> starting cluster");
+        internalCluster().startNodes(3, buildSettings(false));
+
+        // Create an index and distribute it across the 3 nodes
+        createAndPopulateIndex(INDEX_NAME, SHARD_COUNT, REPLICA_COUNT);
+
+        // Create another index so that once we drop the first index, we
+        // can still assert that the cluster is green.
+        createAndPopulateIndex(INDEX_NAME + "-other", SHARD_COUNT, REPLICA_COUNT);
+
+        ensureGreen();
+
+        // This is so that when then node comes back up, we have a dangling index that could
+        // be recovered, but shouldn't be.
+        logger.info("--> restarted a random node and deleting the index while it's down");
+        internalCluster().restartRandomDataNode(new InternalTestCluster.RestartCallback() {
+
+            @Override
+            public Settings onNodeStopped(String nodeName) throws Exception {
+                deleteIndex(INDEX_NAME);
+                return super.onNodeStopped(nodeName);
+            }
+        });
+
+        ensureGreen();
+
+        assertFalse("Did not expect dangling index " + INDEX_NAME + " to be recovered", indexExists(INDEX_NAME));
+    }
+
+    private void createAndPopulateIndex(String name, int shardCount, int replicaCount) throws InterruptedException {
+        logger.info("--> creating test index: {}", name);
+        assertAcked(
+            prepareCreate(
+                name,
+                Settings.builder()
+                    .put("number_of_shards", shardCount)
+                    .put("number_of_replicas", replicaCount)
+                    .put(Store.INDEX_STORE_STATS_REFRESH_INTERVAL_SETTING.getKey(), 0)
```

Comment:
why is this necessary?

Reply:
Ahem...that's copy-and-paste. Can you tell it's the first time I've written an ES IT? 😅